### PR TITLE
Added hook to add entity spectator shader

### PR DESCRIPTION
**Pull request by iLexiconn**

### CODE REVIEW

**Comment by iLexiconn** [comment]
Compacted the patch. I couldn't use forgeHooksclient since `loadshader` is private.

File: patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch [hunk 1]
```diff
@@ -1,5 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/client/renderer/EntityRenderer.java
 +++ ../src-work/minecraft/net/minecraft/client/renderer/EntityRenderer.java
+@@ -199,7 +199,7 @@
+             else if (p_175066_1_ instanceof EntityEnderman)
+             {
+                 this.func_175069_a(new ResourceLocation("shaders/post/invert.json"));
+-            }
++            } else {ResourceLocation r = net.minecraftforge.fml.client.registry.ClientRegistry.getEntityShader(p_175066_1_.getClass()); if (r != null) this.func_175069_a(r);}
```
Comment (Lunatrius) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: This was fine as it was before - the patch is now actually larger (one removal and one addition, whereas it was just an addition before). The changes to the line numbers in the hunks are just an unfortunate side effect.

Comment (LexManos) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: And thus is why we use braces on new lines, it allows for better patches.

Comment (iLexiconn) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: Welp, @diesieben07 told me to change my patch to this. I'll change it back I guess.

**Comment by LexManos** [comment]
You could make it public.We have ATs for a reason.

**Comment by iLexiconn** [comment]
Fixed everything

File: patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch (hunk 1 above)
Comment (diesieben07) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: I could have sword somebody said the line numbers changes are not good before...
Oh well. Sorry.

Comment (ShetiPhian) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: @diesieben07 Same here, thus I always replace when I can.
If this is not the case I'll need to change a few lines in my open pull requests.

**Comment by LexManos** [comment]
All that's left is for our rendering guy who actually knows what this is and what it's for to give it the thumbs up.

File: patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch (hunk 1 above)
Comment (LexManos) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: Committing patches that are JUST line number changes is bad.
Or when things are 1 line change and 50 line number changes.
But this one is fine.

**Pull request merged**